appendix-basic65.tex

```diff
@@ -3305,8 +3306,13 @@ \subsection{DT\$}
                 "DD-MON-YYYY", for example: "04-APR-2021".
 \item[Remarks:] {\bf DT\$} is a reserved system variable.
                 For more information on how to set the Real-Time Clock,
-                refer to the Configuration Utility section on page
-                \pageref{sec:configuration-utility}.
+                refer to
+\ifdefined\printmanual
+the {\bf MEGA65 Book}
+\else
+ \bookvref{sec:configuration-utility}
+\fi
+				.
```

Comment:
This is fixing #519 and therefore many places in the published PDFs where `??` instead of correct references are present. The `if` statement decides whether this goes into the full book in which case a textual reference saying `the Mega65 book` and the full name of chapter/section is stated. Unfortunately the text appears as a non functioning link. Else we have an individual PDF like the one for BASIC in this case. As such it makes an internal link which is even clickable.

Please be aware that this code is not by me. In fact there are a number of such links in the whole project. I only adapted the code to fix all but one reference. The remainder you may find in the issue.

I will not respond any more of your comment asking for this and only type #519 as the reference of what was the cause.

If you revert the `??` simply remain unfixed. See #476 and #519 to get why I had become active here.

Reply:
A bit confused on this description you gave on this:

> The if statement decides whether this goes into the full book in which case a textual reference saying the Mega65 book and the full name of chapter/section is stated.

My recollection was that "ifdefined\printmanual" was introduced in preparation for the printing of the user manual prior to the batch 1 release, as prior to printing, we wanted to removed any ?? bad links. I.e. if this is the printed manual (i.e. the user manual), then this link should just say "The MEGA65 book". Otherwise, let it be a direct link to that chapter/section.

It sounds like you've interpreted the purpose of"ifdefined\printmanual" as to do both things, say "mega65 manual" and also give a direct link decription, which I don't think is the case.

Ok, I've had a read of #519 and get your intentions relating to other pdf's within the suite (apart from user-guide and mega65-book), though I get the feeling maybe original intent behind the "ifdefined\printmanual" in a different way.

Anyways, just to assure, I'm not saying all this to disparrage your efforts, which I do appreciate them, it's more that I see a chance to share some of the backstory that I was made aware of by Bit Shifter as he, impakt and myself were working on proof-reading preparations prior to printing. So please see this just a chance for me to share the backstory on these matters that I was aware of, for me to see your perspective on this too, and for us to get on the same page on this.

My availability will be choppy in the coming days as I'm on holidays, but just thought I'd squeeze thesr thoughts in before the flight :-)

Reply:
@gurcei Thank you for your reply.

> My recollection was that "ifdefined\printmanual" was&hellip;

Whenever I approach uncommon land as I did coming into this Latex project instead of assumption I simply followed the `\def` to where this flag is actually set. I did not find but one file `mega65-userguide.tex` where this is set. So I didn't care what PDFs were used for printing but did see - and this is exactly what you are describing - yes, the links from all small PDF are simply none but a text reference. I would have no idea on how different to approach this. And I also see that many of such references exist throuout the project.

The name `\printmanual` confuses me slightly. I thought not the fat book was printed but some of the smaller ones. But it doesn't affect my work here because I see all working properly.

Let me ask you: If you go to https://files.mega65.org/manuals-upload/ and download all PDFs there, how many dead references `??` you find across all of them? I think about six or seven. Now go to your local make env and count again, right only one left. This one I'm asking in https://github.com/MEGA65/mega65-user-guide/issues/476#issuecomment-1363069668

As you go on a holiday please let this pull request open until you find time for a deep dive. I'm happy waiting and wish you a very good time.

Reply:
No probs, we can keep this PR open continue to share thoughts on this, thanks for your patience on this too.

And I agree the name `\printmanual` is a bit of a vague descriptor, it probably should have been named something like `\printeduserguide`, so as to better describe its original purpose.

And yep, agree that the other smaller pdfs had a lot of the '??' bad links too, but weren't on our radar on this iteration.

So yep, I see how you were able to make use of the `\printmanual` in the pdfs to good result, make the '??' disappear for the other pdfs too, just one downside is that it loses some of its original userguide-specific nature, and the end result is less-optimal for the userguide (some links that would have been valid will now say 'the mega65 book' instead).

But no probs, it was still great to identify the points of concern across all the pdf's, and hopefully in the future, we can devise a smarter 'if' on these points, something akin to:

```
if the chapter/section linked to resides within this book:
  then make that direct link
else:
  just display 'the mega65 book'
```

Reply:
@gurcei I see now, what bothers you. Just to make sure for later: The `if` statement is the other way round. The only thing in my PR not working is that `just display 'the mega65 book'` is actually giving an unusable link. For the moment I thought this to be non-fatal much in contradiction to `??` helping actually nothing.

Enjoy your days off.

Reply:
No probs man. Was thinking, we can just leave these `\ifdef\printmanual` checks as they are in this PR, and I've made ticket #522 for any future efforts we can make to make such link-checking smarter.